bentoml/_internal/io_descriptors/image.py

```diff
@@ -190,20 +216,66 @@ def openapi_responses(self) -> OpenAPIResponse:
     async def from_http_request(self, request: Request) -> ImageType:
         content_type, _ = parse_options_header(request.headers["content-type"])
         mime_type = content_type.decode().lower()
+
+        bytes_: bytes | str | None = None
+
         if mime_type == "multipart/form-data":
             form = await request.form()
-            bytes_ = await next(iter(form.values())).read()
-        elif mime_type.startswith("image/") or mime_type == self._mime_type:
+
+            found_mimes: list[str] = []
+
+            for val in form.values():
+                val_content_type = val.content_type  # type: ignore (bad starlette types)
+                if isinstance(val, UploadFile):
+                    found_mimes.append(val_content_type)
+
+                if self._allowed_mimes is None:
+                    if (
+                        val_content_type in MIME_EXT_MAPPING
+                        or val_content_type.startswith("image/")
+                    ):
+                        bytes_ = await val.read()
+                        break
+                elif val_content_type in self._allowed_mimes:
+                    bytes_ = await val.read()
+                    break
+            else:
+                if len(found_mimes) == 0:
+                    raise BadInput("no image file found in multipart form")
+                else:
+                    if self._allowed_mimes is None:
+                        raise BadInput(
+                            f"no multipart image file (with mime type in {MIME_EXT_MAPPING.keys()} or 'image/*'), got files with content types {', '.join(found_mimes)}"
+                        )
+                    else:
+                        raise BadInput(
+                            f"no multipart image file (with mime type in {self._allowed_mimes}), got files with content types {', '.join(found_mimes)}"
+                        )
+
+        elif self._allowed_mimes is None:
+            if mime_type in MIME_EXT_MAPPING or mime_type.startswith("image/"):
+                bytes_ = await request.body()
+        elif mime_type in self._allowed_mimes:
             bytes_ = await request.body()
         else:
-            raise BadInput(
-                f"{self.__class__.__name__} should get `multipart/form-data`, "
-                f"`{self._mime_type}` or `image/*`, got {content_type} instead"
-            )
+            if self._allowed_mimes is None:
+                raise BadInput(
+                    f"no multipart image file (with mime type in {MIME_EXT_MAPPING.keys()} or 'image/*'), got request with content type {mime_type}"
+                )
+            else:
+                raise BadInput(
+                    f"no multipart image file (with mime type in {self._allowed_mimes}), got a request with content type {mime_type}"
+                )
```

Comment:
Are they still "multipart" image files under the `else` condition?

Reply:
Oops, I failed to update this copy-paste.